app/assets/javascripts/discourse/app/mixins/composer-upload-uppy.js

```diff
@@ -0,0 +1,468 @@
+import Mixin from "@ember/object/mixin";
+import { deepMerge } from "discourse-common/lib/object";
+import UppyChecksum from "discourse/lib/uppy-checksum-plugin";
+import UppyMediaOptimization from "discourse/lib/uppy-media-optimization-plugin";
+import Uppy from "@uppy/core";
+import DropTarget from "@uppy/drop-target";
+import XHRUpload from "@uppy/xhr-upload";
+import { warn } from "@ember/debug";
+import I18n from "I18n";
+import { next, run } from "@ember/runloop";
+import getURL from "discourse-common/lib/get-url";
+import { clipboardHelpers } from "discourse/lib/utilities";
+import { observes, on } from "discourse-common/utils/decorators";
+import {
+  bindFileInputChangeListener,
+  displayErrorForUpload,
+  getUploadMarkdown,
+  validateUploadedFile,
+} from "discourse/lib/uploads";
+import { cacheShortUploadUrl } from "pretty-text/upload-short-url";
+
+// Note: This mixin is used _in addition_ to the ComposerUpload mixin
+// on the composer-editor component. It overrides some, but not all,
+// functions created by ComposerUpload. Eventually this will supplant
+// ComposerUpload, but until then only the functions that need to be
+// overridden to use uppy will be overridden, so as to not go out of
+// sync with the main ComposerUpload functionality by copying unchanging
+// functions.
+//
+// Some examples are uploadPlaceholder, the main properties e.g. uploadProgress,
+// and the most important _bindUploadTarget which handles all the main upload
+// functionality and event binding.
+//
+export default Mixin.create({
+  @observes("composer.uploadCancelled")
+  _cancelUpload() {
+    if (!this.get("composer.uploadCancelled")) {
+      return;
+    }
+    this.set("composer.uploadCancelled", false);
+    this.set("userCancelled", true);
+
+    this._uppyInstance.cancelAll();
+  },
+
+  @on("willDestroyElement")
+  _unbindUploadTarget() {
+    this.messageBus.unsubscribe("/uploads/composer");
+
+    if (this.uploadButton && this.uploadButtonEventListener) {
+      this.uploadButton.removeEventListener(
+        "click",
+        this.uploadButtonEventListener
+      );
+    }
+
+    if (this.fileInputEventListener && this.fileInputEl) {
+      this.fileInputEl.removeEventListener(
+        "change",
+        this.fileInputEventListener
+      );
+    }
+
+    if (this.pasteEventListener && this.element) {
+      this.element.removeEventListener("paste", this.pasteEventListener);
+    }
+
+    this.appEvents.off("composer:add-files", this._addFiles.bind(this));
+
+    if (this._uppyInstance) {
+      this._uppyInstance.cancelAll();
```

Comment:
I saw there was a close() function in uppy doc, did you look at using it ?

Reply:
Ah yes in this case `close()` would be better, thanks